js/src/secureApi.js

```diff
@@ -20,122 +20,109 @@ const sysuiToken = window.localStorage.getItem('sysuiToken');
 
 export default class SecureApi extends Api {
   constructor (url, nextToken) {
-    super(new Api.Transport.Ws(url, sysuiToken));
+    super(new Api.Transport.Ws(url, sysuiToken, false));
 
+    this._url = url;
     this._isConnecting = true;
-    this._connectState = sysuiToken === 'initial' ? 1 : 0;
     this._needsToken = false;
+
     this._dappsPort = 8080;
     this._dappsInterface = null;
     this._signerPort = 8180;
-    this._followConnectionTimeoutId = null;
 
     // Try tokens from localstorage, then from hash
-    this._tokensToTry = [ sysuiToken, nextToken ].filter((t) => t && t.length);
+    this._tokens = [
```

Comment:
What about this case - https://github.com/ethcore/parity/issues/3620

(In this case there is a sysuiToken, however we need to try with `initial` to generate a token. Basically would try `nextToken || 'initial'` on line 36, i.e. in this case these is a syuiToken, but won't be a nextToken - first would fail, second would succeed)